contracts/ERC20SimpleSwap.sol

```diff
@@ -173,18 +173,18 @@ contract ERC20SimpleSwap {
     paidOut[beneficiary] = paidOut[beneficiary].add(totalPayout);
     totalPaidOut = totalPaidOut.add(totalPayout);
     /* do the actual payments */
-
-    require(token.transfer(recipient, totalPayout.sub(callerPayout)), "SimpleSwap: SimpleSwap: transfer failed");
-    /* do a transfer to the caller if specified*/
-    if (callerPayout != 0) {
-      require(token.transfer(msg.sender, callerPayout), "SimpleSwap: SimpleSwap: transfer failed");
-    }
     emit ChequeCashed(beneficiary, recipient, msg.sender, totalPayout, cumulativePayout, callerPayout);
     /* let the world know that the issuer has over-promised on outstanding cheques */
     if (requestPayout != totalPayout) {
       bounced = true;
       emit ChequeBounced();
     }
+
+    require(token.transfer(recipient, totalPayout.sub(callerPayout)), "transfer failed");
```

Comment:
We can remove the `sub` operation if we put this in an `else` branch from `if (callerpayout != 0)`. This will reduce the gas costs slightly if there is no `callerPayout` and then copy this line inside the `if` statement`

Reply:
done